src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -190,10 +200,34 @@ class EthService(
   }
 
   def submitHashRate(req: SubmitHashRateRequest): ServiceResponse[SubmitHashRateResponse] = {
-    //todo do we care about hash rate for now?
+    hashRate.updateAndGet(new UnaryOperator[Map[ByteString, (BigInt, Date)]] {
+      override def apply(t: Map[ByteString, (BigInt, Date)]): Map[ByteString, (BigInt, Date)] = {
+        val now = new Date
+        clearTimeouts(now, t + (req.id -> (req.hashRate, now)))
```

Comment:
Yes, documentation says... ;)
Actually there is no conflict here, because the doc specifies strict rules on the input, and we're simply relaxing them (just like Geth, for convenience of manual interaction).  The input accepted by us is a superset of what's accepted in the doc, without any ambiguity.